packages/shared/src/components/modals/EditWelcomePostModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { MouseEvent, ReactElement, useState } from 'react';
+import { useMutation, useQueryClient } from 'react-query';
+import { Modal } from './common/Modal';
+import { Button } from '../buttons/Button';
+import { NewCommentModalProps } from './NewCommentModal';
+import { PromptOptions, usePrompt } from '../../hooks/usePrompt';
+import PreviewTab from './PreviewTab';
+import { editPost, PostData } from '../../graphql/posts';
+import ContentWriteTab from './ContentWriteTab';
+import { getPostByIdKey } from '../../hooks/usePostById';
+import { useToastNotification } from '../../hooks/useToastNotification';
+
+enum EditPostTab {
+  Write = 'Write',
+  Preview = 'Preview',
+}
+
+const promptOptions: PromptOptions = {
+  title: 'Discard changes',
+  description: 'Are you sure you want to close and discard your changes?',
+  okButton: {
+    className: 'btn-primary-ketchup',
+    title: 'Discard',
+  },
+  cancelButton: {
+    title: 'Stay',
+  },
+};
+
+function EditWelcomePostModal({
```

Comment:
Eventually in v6 you probably suggest to make this generic right?
For now its fine like this.

Reply:
In v6 we will have a whole new page, right?